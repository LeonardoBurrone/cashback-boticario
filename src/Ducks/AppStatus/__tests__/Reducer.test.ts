import {
  changeIsLoggedInAction,
  changeLoadingAction,
  changeLoadingMessageAction,
  changeRequestErrorAction
} from '../Actions';
import appStatusReducer, { initialState } from '../Reducer';

describe('Test AppStatus Reducer', () => {
  it('Check the initial state', () => {
    expect(appStatusReducer(initialState, {})).toMatchSnapshot();
  });

  it('Check reducer state after "changeIsLoggedInAction" was dispatched', () => {
    expect(appStatusReducer(initialState, changeIsLoggedInAction(true))).toMatchSnapshot();
  });

  it('Check reducer state after "changeLoadingAction" was dispatched', () => {
    expect(appStatusReducer(initialState, changeLoadingAction(true))).toMatchSnapshot();
  });

  it('Check reducer state after "changeLoadingMessageAction" was dispatched', () => {
    expect(appStatusReducer(initialState, changeLoadingMessageAction('Loading message...'))).toMatchSnapshot();
  });

  it('Check reducer state after "changeRequestErrorAction" was dispatched', () => {
    expect(appStatusReducer(initialState, changeRequestErrorAction(true))).toMatchSnapshot();
  });
});
