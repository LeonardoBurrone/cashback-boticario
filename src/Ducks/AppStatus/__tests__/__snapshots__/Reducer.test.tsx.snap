// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AppStatus Reducer Check reducer state after "changeIsLoggedInAction" was dispatched 1`] = `
Object {
  "isLoggedIn": true,
  "loading": false,
  "loadingMessage": "",
  "requestFailed": false,
}
`;

exports[`Test AppStatus Reducer Check reducer state after "changeLoadingAction" was dispatched 1`] = `
Object {
  "isLoggedIn": false,
  "loading": true,
  "loadingMessage": "",
  "requestFailed": false,
}
`;

exports[`Test AppStatus Reducer Check reducer state after "changeLoadingMessageAction" was dispatched 1`] = `
Object {
  "isLoggedIn": false,
  "loading": false,
  "loadingMessage": "Loading message...",
  "requestFailed": false,
}
`;

exports[`Test AppStatus Reducer Check reducer state after "changeRequestErrorAction" was dispatched 1`] = `
Object {
  "isLoggedIn": false,
  "loading": false,
  "loadingMessage": "",
  "requestFailed": true,
}
`;

exports[`Test AppStatus Reducer Check the initial state 1`] = `
Object {
  "isLoggedIn": false,
  "loading": false,
  "loadingMessage": "",
  "requestFailed": false,
}
`;
